### DISCUSSION TITLE
Archway Inflationary Rewards

### DISCUSSION THREAD

**Discussion by zanicar** [archway-network/discussions#10] [Archway Blog]
![Blog Banner](https://github.com/archway-network/discussions/blob/main/media/nasa-rTZW4f02zY8-unsplash-cropped.jpg?raw=true)
Photo by <a href="https://unsplash.com/@nasa?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">NASA</a> on <a href="https://unsplash.com/photos/rTZW4f02zY8?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Unsplash</a>

## Background

Archway launched as the Value Capture Chain with its innovative Value Capture Engine that enables sustainable economic models. This Value Capture Engine has 3 major components, namely:

1. Smart Contract Premiums - a fee set by developers for using their contracts;
2. Inflationary Yield - a portion of the network's token inflation is distributed directly to dapps;
3. Gas Fee Rebates - contracts earn a portion of the gas they use;

Both Inflationary Yield and Gas Fee Rebates are parameterized, meaning the Archway community may change this through governance. The ratios at launch were set according to results produced by extensive economic research and modeling, as well as the technical limitations of the Cosmos ecosystem and technology stack. These ratios started at 25-75% and 50-50% for inflationary rewards and gas fee rebates between developers and validators respectively.

Note, that according to our economic paper, our intent with gas fee rebates is to distribute 50% to developers and burn the remainder. However, at launch we opted to not implement the burn yet due to security and other technical considerations. As from [Archway v3.0.0](https://github.com/archway-network/archway/releases/tag/v3.0.0), the [Fee Burn feature](https://github.com/archway-network/archway/pull/422) has been introduced... At the time of writing the Fee Burn feature has been deployed to the Constantine testnet (as part of [Archway v4.0.2](https://github.com/archway-network/archway/releases/tag/v4.0.2)) and can be expected as part of a mainnet upgrade in the near future.

## Intent, Current Constraints and the Future

Similarly, our intent to provide a truly innovative and adaptive underlying economics model is premised not only on what is currently available across the Cosmos ecosystem, but also on that which is to come; taking into consideration what efforts, initiatives and endeavors are currently undertaken. Our current implementation attempts to match the ideal as closely as possible using what is available, while we continually work to reach this ideal as the entire ecosystem grows and matures together.

As it stands, we are awaiting the release of ABCI 2.0 to the Cosmos stack (expected in SDK v0.50), as this allows for more feasible and finer granular access to data points required by the envisaged ideal economics model. Dev Inflationary Tokens (DIT - a.k.a. the developer portion of Inflationary Yield) are currently distributed to dapps based on their proportional fill rate of the current block, with the remainder allocated to the Dapp Treasury. This results in a lower effective distribution of DIT to dapps and a larger portion to the Dapp Treasury, compared to distribution based on a portion of the block Optimal Fill Percentage (OFP).

## Our Suggestion

The Dapp Treasury is a developer community pool and the accumulated funds are still available to dapp developers. Our current suggestion is that the Dapp Treasury may be utilized to subsidize the Rewards Withdrawal fees while we work on a protocol enhancement to significantly improve the rewards withdrawal process (see references for details). Ultimately the developer community is responsible for steering and allocation of these funds, so please feel free to make your own suggestions or recommendations as to how the Dapp Treasury should be utilized to best benefit the Archway developer community.

We kindly ask that developers bear with us as we grow both the protocol and the ecosystem towards ultimate sustainability.

## References

- [Archway Web Site](https://archway.io/)
- [Archway Technical Papers](https://archway.io/whitepapers)
- [Archway Economics - A Sustainable Hub for Dapps](https://archway.io/assets/archway-economics.pdf)
- [Archway Protocol Specs - Rewards](https://github.com/archway-network/archway/tree/main/x/rewards/spec)
- [Archway Releases](https://github.com/archway-network/archway/releases)
- [Feature Request: Enhancing Querying and Withdrawal of Rewards](https://github.com/archway-network/archway/issues/434)

**Comment by jlm0** [comment]
Thank you for the suggestion to utilize the Dapp Treasury to subsidize rewards withdrawal fees in the short term. I appreciate any additional clarity on these questions:

- How quickly could the treasury be depleted if used heavily for subsidies? Do we have projections on sustainability?
- How is governance of the dapp treasury currently handled? Is there a transparent process for deciding how funds are utilized?
- Will reliance on the treasury reduce the urgency to implement protocol-level solutions for lower withdrawal fees long term? What are the expected long-term solutions here?

I'm glad the Dapp Treasury exists as an interim option but want to better understand the governance, sustainability, and timeline for long-term solutions.

**Comment by ohadbachner** [comment]
To ensure endless trust in the SubDAO overseeing the Dapp Treasury in Archway, we recommend integrating our Product Bloc, the Mini DAO V2. As detailed in our [article](https://medium.com/chelo-finance/decentralized-autonomous-organizations-made-easy-with-mini-dao-v2-562ef1fce209), the Mini DAO V2 offers:

Simplified Governance: Streamlined proposal and voting mechanisms, making it easier for members to participate.
Flexible Role Assignments: Dynamic role-based access control, allowing for a variety of governance structures.
Transparent Decision-making: All actions and decisions are recorded on-chain, ensuring full transparency.
Efficient Fund Management: Built-in mechanisms for fund allocation, distribution, and management, tailored for community-driven projects.
Modular Control: The most significant advantage is its modular design. The main DAO retains overarching control, allowing it to intervene in the Mini DAO's decisions when necessary. This includes the ability to cancel proposals or refund the treasury, ensuring a safety net against any potential mismanagement.
By adapting the Mini DAO V2's EVM-ready code to be compatible with CosmWasm and Archway, we can leverage these features to ensure a decentralized and trustless management system for the Dapp Treasury. This will not only empower the Archway developer community but also instill greater confidence in the governance process, knowing that there's a mechanism for oversight and intervention when needed.
Let me know what do you think


Ohad

**Reply by zanicar** [reply]
Excellent article, very comprehensive...

So I'll raise the question that needs to be asked: "Has it been audited?"

**Reply by ohadbachner** [reply]
No it is not unfortunately, we dont have enough funds for that
